suite-native/transactions/src/components/TransactionsList/TransactionList.tsx

```diff
@@ -104,49 +113,53 @@ const renderSectionHeader = ({ section: { monthKey } }: RenderSectionHeaderParam
 export const TX_PER_PAGE = 25;
 
 export const TransactionList = ({
-    transactions,
     areTokensIncluded,
     listHeaderComponent,
-    fetchMoreTransactions,
     accountKey,
     tokenContract,
 }: AccountTransactionProps) => {
+    const dispatch = useDispatch();
+    const [isRefreshing, setIsRefreshing] = useState(false);
+
     const {
         applyStyle,
         utils: { colors },
     } = useNativeStyles();
-    const isLoadingTransactions = useSelector(selectIsLoadingTransactions);
-    const dispatch = useDispatch();
-    const [isRefreshing, setIsRefreshing] = useState(false);
+    const isLoadingTransactions = useSelector((state: TransactionsRootState) =>
+        selectIsLoadingAccountTransactions(state, accountKey),
+    );
+    const transactions = useSelector(
+        (state: TransactionsRootState & FiatRatesRootState & SettingsSliceRootState) =>
+            selectAccountOrTokenAccountTransactions(
+                state,
+                accountKey,
+                tokenContract ?? null,
+                areTokensIncluded,
+            ),
+    );
+    const isLoaderVisible = isLoadingTransactions && transactions.length === 0;
 
-    const initialPageNumber = Math.ceil((transactions.length || 1) / TX_PER_PAGE);
-    const [page, setPage] = useState(initialPageNumber);
+    const fetchTransactions = useCallback(() => {
+        return dispatch(fetchAllTransactionsForAccountThunk({ accountKey }));
+    }, [accountKey, dispatch]);
 
     useEffect(() => {
-        // it's okay to hardcode 1 because this is initial fetch and in case transactions are already loaded, nothing will happen anyway
-        // because of the check in fetchMoreTransactionsThunk
-        fetchMoreTransactions(1, TX_PER_PAGE);
-    }, [fetchMoreTransactions]);
-
-    const handleOnEndReached = useCallback(async () => {
-        try {
-            await fetchMoreTransactions(page + 1, TX_PER_PAGE);
-            setPage((currentPage: number) => currentPage + 1);
-        } catch (e) {
-            // TODO handle error state (show retry button or something
-        }
-    }, [fetchMoreTransactions, page]);
+        fetchTransactions();
+    }, [fetchTransactions]);
 
     const handleOnRefresh = useCallback(async () => {
         try {
             setIsRefreshing(true);
-            await dispatch(fetchAndUpdateAccountThunk({ accountKey }));
+            await Promise.allSettled([
+                dispatch(fetchAndUpdateAccountThunk({ accountKey })),
+                fetchTransactions(),
+            ]);
         } catch (e) {
             // Do nothing
         }
         // It's usually too fast so loading indicator only flashes for a moment, which is not nice
         setTimeout(() => setIsRefreshing(false), 1500);
```

Comment:
can you handle timeout in destructor, please?

Reply:
I think it's not necessary here, it should not produce any incosistent state and it will be too much work to handle it :D